$NetBSD: patch-ac,v 1.8 2005/10/07 12:18:19 rillig Exp $

The original commit message for this patch:

    http://mail-index.netbsd.org/source-changes/1999/09/17/0023.html

    Patch from Juergen Hannken-Illjes, which should solve PR#8308 (net 0 is
    reserved for lo0, it can't be used on other interfaces).

Here is PR #8308:

    http://www.NetBSD.org/cgi-bin/query-pr-single.pl?number=8308

and also the change that exposed the bug:

    http://mail-index.netbsd.org/source-changes/1999/08/24/0035.html

and finally, the discussion referred to in the above change:

    http://mail-index.netbsd.org/tech-net/1999/06/30/0002.html
    http://mail-index.netbsd.org/tech-net/1999/07/01/0005.html
    http://mail-index.netbsd.org/tech-net/1999/07/01/0007.html
    http://mail-index.netbsd.org/tech-net/1999/07/02/0003.html
    http://mail-index.netbsd.org/tech-net/1999/07/02/0016.html
    http://mail-index.netbsd.org/tech-net/1999/07/02/0017.html
    http://mail-index.netbsd.org/tech-net/1999/07/02/0018.html
    http://mail-index.netbsd.org/tech-net/1999/07/02/0020.html
    http://mail-index.netbsd.org/tech-net/1999/07/02/0022.html
    http://mail-index.netbsd.org/tech-net/1999/07/03/0004.html

--- etc/atalkd/main.c.orig	Sat Oct  5 08:20:13 2002
+++ etc/atalkd/main.c
@@ -278,12 +278,12 @@ static void as_timer(int sig)
 		    LOG(log_info, logtype_atalkd, "config for no router" );
 		      
 		    if ( iface->i_flags & IFACE_PHASE2 ) {
-			iface->i_rt->rt_firstnet = 0;
+			iface->i_rt->rt_firstnet = htons(1);
 			iface->i_rt->rt_lastnet = htons( STARTUP_LASTNET );
 			setaddr( iface, IFACE_PHASE2,
 				iface->i_addr.sat_addr.s_net,
 				iface->i_addr.sat_addr.s_node,
-				0, htons( STARTUP_LASTNET ));
+				htons(1), htons( STARTUP_LASTNET ));
 		    }
 		    if ( looproute( iface, RTMP_ADD ) ) { /* -1 or 1 */
 			LOG(log_error, logtype_atalkd,
